content/blog/0021-bundle-accounting.mdx

```diff
@@ -0,0 +1,163 @@
+
+A couple of weeks ago I inherited a `create-react-app` project, yes, shudder.
+It had all the classics, AnyScript, class components, esoteric 1-off deps preventing updates,
+a million lint violations of the `==` kind, multiple overlapping component libraries,
+and multiple style solutions.
+
+My task was to add 4 pages and a whole bunch of features.
+Using the existing material was out of the question, I **had** to create my own lib from scratch
+if I wanted to have any semblence of confidence in the app.
+
+I started hacking at it. It didn't take a full hour before I had to wrangle untyped garbage from the legacy
+side of the fence. It was an http call, it wasn't parsed, there was no `zod` or something. It wasn't
+even typed `Promise<Something>`, it was proudly `Promise<any>` and then just consumed like thing.
+I wasn't there at the dawn of creation, I had no idea what the previous dev thought the shape was
+I could only follow function names and see the properties being arbitrarily accessed. So I did research to
+find out what the shape was supposed to be and created a util:
+
+```ts
+const UnsafeTypeId: unique symbol = Symbol.for("@types/Unsafe");
+
+/**
+ * Brands `A` as an unsafe type.
+ * An unsafe type is a type that was cast but never validated.
+ *
+ * @example
+ * ```ts
+ * const response = await axios.get(...)
+ *
+ * const data = response.data as Unsafe<Todo>
+ * ```
+ *
+ * @author Datner<contact@datner.me>
+ */
+export type Unsafe<A> = A & {
+  readonly [UnsafeTypeId]?: "Unsafe";
+};
+```
+
+Now I can give a proper shape and acknowledge that it is infact a bullshit promise, I never made sure that
+The shape is actually that, but it's more than `any`.
+Then I encountered another situation that prompted a different util, and then another, and then
+even more, I found myself building all the
+inftrastructure and utils for the basic essentials of just making usable software.
+I tried coping, letting go, but it just kept coming back and it dawned on me. I'm just reimplemeting effect.
+
+Worse, I'm reimplemeting effect by trying to hot-glue libs together. Oh the axios client needs the current
+tenant so it needs access to the context but it can change in the lifetime of the application and it also
+needs the bearer token but it might need revalidation or even a relog oh but I make too many requests
+for the same thing I need some caching--- It's been nearly a month
+now and I've only been building inftrastructure!
+The boring kind too! Plus I hate the result and it's full of bugs I won't see until they pop up in production!
+
+I gave up. Whatever the effect tax is, it's worth it.
+From what we've seen, the effect tax is about 50kb. Depending on what you feel this can be
+a lot or very little, but considering that I added around 20k LOC to the project, plus new dependencies
+It was not worth considering.
```

Comment:
```suggestion
So I gave up. Whatever the `effect` "tax" is, it's worth it. From what we've seen, the "tax" on bundle size is about 50kb. Depending on your use-case, this may seem like a lot or very little. But considering that I added around 20k LOC to the project, as well as some new dependencies, for me it wasn't even a consideration.
```